cli/uninstall/uninstall.go

```diff
@@ -26,53 +26,56 @@ const (
 )
 
 // remove removes binary/directory and symlinks from directory.
+// It returns (true if symlink was removed, error).
 func remove(program string, programVersion string, directory string,
-	cmdCtx *cmdcontext.CmdCtx) error {
+	cmdCtx *cmdcontext.CmdCtx) (bool, error) {
 	var linkPath string
 	var err error
 
 	if program == search.ProgramCe || program == search.ProgramEe {
 		if linkPath, err = util.JoinAbspath(directory, "tarantool"); err != nil {
-			return err
+			return false, err
 		}
 	} else {
 		if linkPath, err = util.JoinAbspath(directory, program); err != nil {
-			return err
+			return false, err
 		}
 	}
 
 	if _, err := os.Stat(directory); os.IsNotExist(err) {
-		return fmt.Errorf("couldn't find %s directory", directory)
+		return false, fmt.Errorf("couldn't find %s directory", directory)
 	} else if err != nil {
-		return fmt.Errorf("there was some problem with %s directory", directory)
+		return false, fmt.Errorf("there was some problem with %s directory", directory)
 	}
 
 	fileName := program + version.FsSeparator + programVersion
 	path := filepath.Join(directory, fileName)
 
 	if _, err := os.Stat(path); os.IsNotExist(err) {
-		return fmt.Errorf("there is no %s installed", program)
+		return false, fmt.Errorf("there is no %s installed", program)
 	} else if err != nil {
-		return fmt.Errorf("there was some problem locating %s", path)
+		return false, fmt.Errorf("there was some problem locating %s", path)
 	}
 	// Get path where symlink point.
 	resolvedPath, err := util.ResolveSymlink(linkPath)
 	if err != nil {
-		return fmt.Errorf("failed to resolve symlink %s: %s", linkPath, err)
+		return false, fmt.Errorf("failed to resolve symlink %s: %s", linkPath, err)
 	}
+	var isSymlinkRemoved bool
 	// Remove symlink if it points to program.
 	if strings.Contains(resolvedPath, fileName) {
 		err = os.Remove(linkPath)
 		if err != nil {
-			return err
+			return false, err
 		}
+		isSymlinkRemoved = true
 	}
 	err = os.RemoveAll(path)
 	if err != nil {
-		return err
+		return isSymlinkRemoved, err
```

Comment:
What will happen if symlink is removed but os.RemoveAll(path) will fail? Symlink will be removed but program won't.
Should we handle this case somewhat different?

Reply:
In this case, uninstall switching won't happen, and `tt` will print error, i.e., the behavior will not change.